src/main/java/cn/nukkit/player/handler/LoginPacketHandler.java

```diff
@@ -142,4 +125,17 @@ public void onCompletion(Server server) {
         session.sendPacket(this.server.getPackManager().getPacksInfos());
         return true;
     }
+
+    private boolean isValidName(String username) {
+        int length = username.length();
+
+        if (length > 16 || length < 3) {
+            return false;
+        }
+
+        Pattern pattern = Pattern.compile("^[aA-zZ\\s\\d_]+$");
```

Comment:
why compile the regex every time? compile it once and put it in a `private static final` field

Reply:
Done.

Reply:
You should make it static in my opinion

Reply:
and maybe include rcon and console restrictions into the regex too

Reply:
> and maybe include rcon and console restrictions into the regex too

I did as you said. Check out the changes.